src/superqt/utils/_errormsg_context.py

```diff
@@ -0,0 +1,145 @@
+        The title of the `QMessageBox`, by default "An error occurred"
+    msg_template : str, optional
+        The message to show in the `QMessageBox`. The message will be formatted
+        using three variables:
+        - `exc_value`: the exception instance
+        - `exc_type`: the exception type
+        - `tb`: the traceback as a string
+        The default template is the content of the exception: `"{exc_value}"`
+    buttons : QMessageBox.StandardButton, optional
+        The buttons to show in the `QMessageBox`, by default
+        `QMessageBox.StandardButton.Ok`
+    parent : QWidget | None, optional
+        The parent widget of the `QMessageBox`, by default `None`
+    use_error_message : bool | QErrorMessage, optional
+        Whether to use a `QErrorMessage` instead of a `QMessageBox`. By default
+        `False`. `QErrorMessage` shows a checkbox that the user can check to
+        prevent seeing the message again (based on the text of the formatted
+        `msg_template`.) If `True`, the global `QMessageError.qtHandler()`
+        instance is used to maintain a history of dismissed messages. You may also pass
+        a `QErrorMessage` instance to use a specific instance. If `use_error_message` is
+        True, `parent` is ignored.
+
+    Attributes
+    ----------
+    widget : QMessageBox | None
+        The `QMessageBox` instance that was created. If `use_error_message` is
+        True, this will be `None`.
+    exception : BaseException | None
+        Will hold the exception instance if an exception was raised and caught. This
+        can be used to determine whether an exception was raised or not.
+
+    Example
+    -------
+    ```python
+    from qtpy.QtWidgets import QApplication
+    from superqt.utils import exceptions_as_dialog
+
+    app = QApplication([])
+
+    with exceptions_as_dialog() as ctx:
+        raise Exception("This will be caught and shown in a QMessageBox")
+
+    # you can access the exception instance here
+    assert ctx.exception is not None
+
+    # with exceptions_as_dialog(ValueError):
+    #     1 / 0  # ZeroDivisionError is not caught, so this will raise
+
+    with exceptions_as_dialog(msg_template="Error: {exc_value}"):
+        raise Exception("This message will be inserted at 'exc_value'")
+
+    for _i in range(3):
+        with exceptions_as_dialog(AssertionError, use_error_message=True):
+            assert False, "Uncheck the checkbox to ignore this in the future"
+
+    app.exec()  # needed only for the use_error_message example to show
+    ```
+    """
+
+    widget: QMessageBox | None
+    exception: BaseException | None
+
+    def __init__(
+        self,
+        exceptions: type[BaseException] | tuple[type[BaseException], ...] = Exception,
+        icon: QMessageBox.Icon = QMessageBox.Icon.Critical,
+        title: str = "An error occurred",
+        msg_template: str = "{exc_value}",
+        buttons: QMessageBox.StandardButton = QMessageBox.StandardButton.Ok,
```

Comment:
If we would like to allow select buttons, there should be an option to check what users select. So save the `self.widget.exec()` result.

Reply:
yeah, i had made a little note in the `__exit__` method to that effect.   Do you remember whether the  Dialog itself stores the last result of exec?  Or not in the case of multiple buttons?  I can just add another attribute

Reply:
`dialog.result()`

Reply:
ah, ok... so, what do you think?

```python
with exceptions_as_dialog() as ctx:
    ...
ctx.dialog.result()
```

or 
```python
with exceptions_as_dialog() as ctx:
    ...
ctx.exec_result()
```

Reply:
i think i prefer the former... it would also allow `ctx.dialog.clickedButton()` for example